rust_src/src/data.rs

```diff
@@ -597,6 +598,29 @@ pub unsafe extern "C" fn store_symval_forwarding(
     }
 }
 
+unsafe fn update_buffer_defaults(objvar: *const LispObject, newval: LispObject) {
+    // If this variable is a default for something stored
+    // in the buffer itself, such as default-fill-column,
+    // find the buffers that don't have local values for it
+    // and update them.
+    let defaults = LispBufferRef::new(&mut buffer_defaults as *mut Lisp_Buffer);
+    let defaults_as_object = defaults.as_ptr() as *const LispObject;
+    if objvar > defaults_as_object && objvar < defaults_as_object.add(1) {
+        let offset = (objvar as *const c_char).offset_from(defaults.as_ptr() as *const c_char);
+        let idx = per_buffer_idx(offset);
```

Comment:
Here we compute an offset using a `*c_char`, and then pass it to `per_buffer_idx` which adds it to a pointer of type `*LispBuffer`. That'll give the wrong result, because offsets are measured relative to the size of the item pointed to.

Reply:
Good catch. I will give this code a closer look.

Reply:
I should probably have been clearer about this; I want the offset to still be the count of LispObjects, not the count of bytes. It reduces the amount of casting involved, and it will make it easier to change this in the future.

Reply:
Ah, I was not 100% sure we could trust it. I will give that a go.